src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -215,6 +226,15 @@ public async Task InitializeUIAffinitizedServicesAsync(IAsyncServiceProvider asy
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_threadingContext.DisposalToken);
 
             openFileTracker.ProcessQueuedWorkOnUIThread();
+
+            // Switch to a background thread to avoid loading option providers on UI thread (telemetry is reading options).
```

Comment:
@tmat Where is the reading of option providers happening in there? I see that the telemetry service has options which it's passing through to the EtwLogger, but that's not actually going to be doing any fetching on this path I'd imagine. I see the code in the stack on the internal bug for the FileLogger, but how is that being invoked here?

Reply:
InitializeTelemetrySession calls VisualStudioWorkspaceTelemetryService.CreateLogger, which calls FileLogger ctor, which reads options.

Reply:
Ah, my misunderstandng was our internal code index was out of date, and didn't have the FileLogger constructor call. This is fine.